### PR TITLE
tests: add end-to-end store test for classic confinement

### PR DESCRIPTION
**Pull request by mvo5**
Adds an end-to-end test for classic confinement

### CODE REVIEW

File: tests/lib/snaps/test-snapd-classic-confinement/meta/snap.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: classic-confinement
+name: test-snapd-classic-confinement
 version: 1.0
 confinement: classic
 apps:
```
Comment (zyga) [inline tests/lib/snaps/test-snapd-classic-confinement/meta/snap.yaml]: You want to rename the app as well

Comment (mvo5) [inline tests/lib/snaps/test-snapd-classic-confinement/meta/snap.yaml]: Indeed, good catch. Thank you!

File: tests/main/classic-confinement/task.yaml [hunk 2]
```diff
@@ -4,16 +4,25 @@ systems: [-ubuntu-core-16-64, -ubuntu-core-16-arm-64, -ubuntu-core-16-arm-32]
...
-    classic-confinement | MATCH lala
+    test-snapd-classic-confinement | MATCH lala
+    snap remove test-snapd-classic-confinement
+
+    echo "Check that we can install classic confinement snaps from the store"
+    snap install --classic test-snapd-classic-confinement
+    test-snapd-classic-confinement | MATCH lala
+
+    echo "Snap refresh from the store also works (2.0 is in beta, 1.0 in stable)"
+    snap refresh --beta test-snapd-classic-confinement
```
Comment (pedronis) [inline tests/main/classic-confinement/task.yaml]: should we check that it got really an update looking at snap info perhaps ? can be done in a follow up

Comment (mvo5) [inline tests/main/classic-confinement/task.yaml]: :+1:

**Comment by mvo5** [comment]
Looks fine, my comment https://github.com/snapcore/snapd/pull/2606#discussion_r95673734 actually applies here but we can do this in a followup, it looks like we can probably extract a helper but its fine as it is.

File: tests/main/classic-confinement/task.yaml (hunk 2 above)
Comment (mvo5) [inline tests/main/classic-confinement/task.yaml]: I will do a followup for that so that we have this one.

**Pull request merged**